### ISSUE TITLE
Error: Couldn't find preset "module:metro-react-native-babel-preset"

### ISSUE THREAD

**Issue by dylancompanjen** [facebook/react-native#21241]
## Environment
React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: x64 Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
      Memory: 68.43 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 9.8.0 - /usr/local/bin/node
      Yarn: 1.9.4 - /usr/local/bin/yarn
      npm: 6.3.0 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
    IDEs:
      Xcode: 10.1/10O23u - /usr/bin/xcodebuild
    npmPackages:
      react: ^16.5.2 => 16.5.2 
      react-native: ^0.57.0 => 0.57.0 
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-cli: 1.2.0
      react-native-git-upgrade: 0.2.7
      react-native-rename: 2.2.2

## Description
When trying to run "react-native-git-upgrade" it gives the following error: 
`git-upgrade ERR! An error occurred during upgrade:`
`Error: Couldn't find preset "module:metro-react-native-babel-preset" relative to directory "Users/...`

## Reproducible Demo
Try to upgrade from React Native 0.56 to 0.57 with "react-native-git-upgrade".

**Comment by HonzaHK** [comment]
Same here, trying to upgrade from 0.53 to 0.57 with "react-native-git-upgrade" on Ubuntu 18.04 LTS

**Comment by JanErikFoss** [comment]
Same error, 0.57.0-rc.3 to 0.57.0 on Mojave 10.14 Beta

**Comment by canyara** [comment]
`yarn add metro-react-native-babel-preset --dev`
https://github.com/facebook/metro/tree/master/packages/metro-react-native-babel-preset

**Comment by HonzaHK** [comment]
@canyara i have done `yarn add metro-react-native-babel-preset --dev`, but got the same error:

`git-upgrade ERR! Error: Couldn't find preset "module:metro-react-native-babel-preset" relative to directory "/home/hk/Desktop/kovosmid_app"
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:293:19 ...`

**Comment by kelset** [comment]
Then I suggest you try to upgrade your project following the [manual steps we wrote in the changelog](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#updating-to-this-version).

If that still happens for you then please create a sample project that reproduces the issue.

**Comment by Amnesthesia** [comment]
I'm getting this as well, even with metro 0.45.6

```$ yarn why metro
=> Found "metro@0.45.6"
info Has been hoisted to "metro"
info Reasons this module exists
   - Specified in "devDependencies"
   - Hoisted from "metro#metro-config#metro"
info Disk size without dependencies: "1.64GB"
info Disk size with unique dependencies: "11.72GB"
info Disk size with transitive dependencies: "43.06GB"
info Number of shared dependencies: 247
```

**Comment by Amnesthesia** [comment]
I seemed to have solved this by removing **everything** in the `babel.rc` file:

```
# .babel.rc
{
}
```

`react-native-git-upgrade` then tries to re-add this during installation. I kept removing it and re-saving the file, and it continued without crashing. Then I added the contents back.

**Comment by MaxInMoon** [comment]
just a note from the [metro's doc](https://github.com/facebook/metro/tree/master/packages/metro-react-native-babel-preset#metro-react-native-babel-preset):

> If you wish to use a custom Babel configuration by writing a .babelrc file in your project's root directory, you must specify all the plugins necessary to transform your code. React Native does not apply its default Babel configuration in this case. So, to make your life easier, you can use this preset to get the default configuration and then specify more plugins that run before it.

**Comment by sonaye** [comment]
Same issue here when trying to upgrade over the last month.

- `0.56.0` to `0.56.1`, failed, waited for the new release.
- `0.56.0` to `0.57.0`, issue is still there, started a project from scratch on top `0.57.0`.
- `0.57.0` to `0.57.1`, issue is still there.

Already followed the manual steps @kelset, and also updated metro to `0.45.6`.

**Comment by canyara** [comment]
> @canyara I already added metro-react-native-babel-preset (and updated my babelrc). The bug still appears.

In my case, there is incompatible issue react 16.5.2 have with react-native 0.57. Downgrade to React 16.5.1, restart react-native everything is ok

**Comment by canyara** [comment]
@dylancompanjen

**Comment by canyara** [comment]
@dylancompanjen by the way, you need quit the react-native terminal and complete restart react-native if you already install metro-react-native-babel-preset

**Comment by kelset** [comment]
> I followed manual update from the changelog (with fixed react@16.5.0) but react-native-git-upgrade still does the error

Why would you follow the manual steps AND do the git-upgrade? 🤔

**Comment by rottenoats** [comment]
After starting a fresh project I got https://github.com/facebook/react-native/issues/7308 error and tried fixing it with react-native-git-upgrade and got this error.

Been having problems with it since last week. Any solutions. (a manual solution shouldn't be needed imo)

**Comment by kelset** [comment]
> a manual solution shouldn't be needed imo

Why? I never use `react-native-git-upgrade` whenever I update my projects, I prefer to do things manually to have granular control over the changes I apply 🤷‍♂️

Moreover, as discussed [here](https://github.com/react-native-community/discussions-and-proposals/blob/master/core-meetings/2018-08-cli-meeting.md#upgrade-and-react-native-git-upgrade), there is noone currently maintaining that portion of the CLI - so it's always a risk to use it.

**Comment by manosim** [comment]
I am also getting exactly the same error when running the tests on a fresh react-native 0.57.1 project. To reproduce:

### Steps

```
react-native init TestProject
cd TestProject/
touch App.test.js

# Add this to App.test.js
# describe('hello', () => {
#   it('basic', () => {
#     expect(1+1).toEqual(2);
#   });
# });

yarn test
```

### Output

```bash
$ jest --no-cache
 FAIL  ./App.test.js
  ● Test suite failed to run

    Couldn't find preset "module:metro-react-native-babel-preset" relative to directory "/Users/manos/Projects/TestProject"

      at node_modules/babel-core/lib/transformation/file/options/option-manager.js:293:19
          at Array.map (<anonymous>)
      at OptionManager.resolvePresets (node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
      at OptionManager.mergePresets (node_modules/babel-core/lib/transformation/file/options/option-manager.js:264:10)
      at OptionManager.mergeOptions (node_modules/babel-core/lib/transformation/file/options/option-manager.js:249:14)
      at OptionManager.init (node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
      at File.initOptions (node_modules/babel-core/lib/transformation/file/index.js:212:65)
      at new File (node_modules/babel-core/lib/transformation/file/index.js:135:24)
      at Pipeline.transform (node_modules/babel-core/lib/transformation/pipeline.js:46:16)

Test Suites: 1 failed, 1 total
```

Worth mentioning that there is an open related issue on the metro repository: https://github.com/facebook/metro/issues/242

Thought I should share it here since I can't see any activity around this issue.

**Comment by rottenoats** [comment]
What solved my problem in the end, was simply building the project under Xcode and then running it with `react-native run-ios`.

If you're attempting to upgrade your project and your stuck and unable to solve any problem, then this is what I did. Not the best way, but it works.

1. Create a fresh project, with the lastest react-native version
2. Ensure to build it in xcode, then test it with react-native run-ios command
3. Copy your source files into your project
4. Re-add all your dependencies (don't copy your package.json file)
5. Run react-native link
6. Add all required permissions in your Info.plist
7. Run the app and see if everything works.
8. Copy your .git into the project and create a new branch

If you run into some problems then you might have skipped a step or I might have forgotten to mention something.

I ran into a few issues, but that's because I forgot to add a few Babel plugins i was using for mobx.

This worked for me. Took a while, but in the end saved time instead of trying to understand vague error messages.

**Comment by luizvidoto** [comment]
Not so much related to this issue but I had another problem when trying to fresh install RN and maybe it could help someone.
I fresh installed RN using react-native-cli init and tried to run using run-ios command but build failed every time.  
So I changed the Xcode project settings to use Legacy Build and then it worked. But the app didn't launch because it was giving me another error. The error was related to babel and I found a solution here: **https://github.com/jquense/yup/issues/216**

**Comment by thomas-gilbert** [comment]
I am getting the same error, even when creating a brand new project with `react-native init`. All I changed from the base is -

1. Moved tests to its own directory.
2. Added one single test. 
3. Added babel-plugin-module-resolver

Please see the repo here - https://github.com/thomas-gilbert/react-native-metro-issue

React Native Environment Info:
    System:
      OS: macOS Sierra 10.12.6
      CPU: x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
      Memory: 337.80 MB / 16.00 GB
      Shell: 5.2 - /bin/zsh
    Binaries:
      Node: 9.4.0 - /usr/local/bin/node
      Yarn: 1.3.2 - /usr/local/bin/yarn
      npm: 5.6.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 11.1, macOS 10.13, tvOS 11.1, watchOS 4.1
      Android SDK:
        Build Tools: 26.0.2, 27.0.1, 27.0.3
        API Levels: 25, 26, 27
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 9.1/9B55 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.5.0 => 16.5.0
      react-native: 0.57.1 => 0.57.1
    npmGlobalPackages:
      create-react-native-app: 1.0.0
      react-native-git-upgrade: 0.2.7

I have been going through quite a few different issues trying to resolve it and any change I make seems to show me another issue. Any help or insight would be appreciated.

**Comment by rottenoats** [comment]
@thomas-gilbert did you open xcode and try building it from there?

**Comment by jacquesdev** [comment]
Best solution so far, worked for me as well https://github.com/facebook/react-native/issues/21241#issuecomment-423610383

**Comment by thomas-gilbert** [comment]
@rottenoats The issue isn't with the app building or bundling; as that works fine. I get the above issue only when running jest.

**Comment by SpQuyt** [comment]
You can try this:
Change the line of file ".babelrc":
from 
{
  "presets": ["module:metro-react-native-babel-preset"]
}
to 
{
  "presets": ["react-native"]
}

It worked for me.

**Comment by manosim** [comment]
Just tried with the fresh `0.57.2` and unfortunately, it looks like this is still happening on a fresh project. Steps to reproduce in my [previous comment](https://github.com/facebook/react-native/issues/21241#issuecomment-424905717).

**Comment by EleisonC** [comment]
@dylancompanjen so what worked for my case was I added this to my jest config in my package.json 
```
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "transform": {
      "^.+\\.(js)$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
    }
```
Hope it helps

**Comment by TheDevXen** [comment]
I always get error while installing vector icons

`error: bundling failed: Error: Couldn't find preset "module:metro-react-native-babel-preset" relative to directory "/home/....`

I tried all ways to resolve this issue but none of them is working..
I am tired solving this since 2 days

**Comment by msaifa** [comment]
I hope this can help
1. Go to new module folder (ex: node_module/react-native-vector-icons)
2. Open .babelrc
3. replace `"presets": ["module:metro-react-native-babel-preset"]`
4. with ` "presets": ["react-native"]`
5. and rebuild project

trims

**Comment by mathewmorris** [comment]
> @Amnesthesia 's workaround allowed my upgrade to progress a bit further, but eventually `react-native-git-upgrade` re-added the preset to .babelrc file and failed at the same error in the end.
> 
> My solution was to rename `.babelrc` -> `.babelrc2` during upgrade which worked. I restored the `.babelrc` config after the upgrade.
> 
> Happened to me when upgrading `v0.57.0` -> `v0.57.1` and `v0.57.1` -> `v0.57.2`.

Worked beautifully! 👍🏼

**Comment by calcazar** [comment]
Do we have a status on this? I started a new project, brought my JS source in to it, ran jest and I get  Couldn't find preset "module:metro-react-native-babel-preset"

**Comment by rszalski** [comment]
@calcazar Try this

1. Create `babel.config.js` with the following content (basically equivalent to stock `.babelrc`)

```
module.exports = function (api) {
  api.cache(true)

  return {
    presets: ['module:metro-react-native-babel-preset']
  }
}
```
2. Remove babel.rc
3. Run `yarn add --dev babel-jest babel-core@^7.0.0-bridge.0 @babel/core`

**Comment by von322** [comment]
on the babel file .babelrc change 
this 
{
  "presets": ["module:metro-react-native-babel-preset"]
}
for this 
{
  "presets": ["react-native"]
}
it worked for me

**Comment by mikemorris** [comment]
**tl;dr: Please publish a new release of `react-native-git-upgrade` to npm**

A fix for this landed in https://github.com/facebook/react-native/pull/20790, but a new release of `react-native-git-upgrade` including this commit has not yet been published. I was able to successfully upgrade by cloning the react-native repo, running `npm i` in the `react-native-git-upgrade` directory, then from my project running
```
node ../react-native/react-native-git-upgrade/index.js 0.57.2
```

Would also close https://github.com/facebook/react-native/issues/21531 (dupe)

**Comment by Foveluy** [comment]
just simply delete `.babelrc` in ##your## project root. you will save your day. After Everything clean and update done, re-add `.bablerc` with the content before you deleted

**Comment by vitorreis** [comment]
This is still happening with a fresh project using latest react-native version (0.57.4)

## How to reproduce 
1) Create a new react-native project
2) Add a test file e.g.:
```
// App.spec.js
it('renders correctly', () => {
  expect(1).toEqual(1);
});
```
3) `npm test`

**Comment by vitorreis** [comment]
I managed to work around it by deleting the `.babelrc` file
and adding a `babel.config.js`

```
module.exports = function (api) {
  api.cache(true);
   return {
    presets: [
      'module:metro-react-native-babel-preset'
    ]
  };
};
```

Discussed in more detail here: https://github.com/facebook/metro/issues/242#issuecomment-431350665

**Comment by mezod** [comment]
It finally worked yesterday night. I say "it" because I can't tell what fixed it. I tried many things like watchman watch-del-all, npm start -- --reset-cache, etc without any luck rebuilding several times until after no significant change, it just went away. Pretty frustrating :D

**Comment by elewin** [comment]
Should this be closed? I too am having the same problem, and "it worked eventually" doesn't seem like much of a solution.

**Comment by gutoglup** [comment]
The problem persists, when I use the 'module:metro-react-native-babel-preset', the application works but don't work the tests. And I use the preset 'react-native', the application don't work, but the tests works.

**Comment by DLevai94** [comment]
I have the same problem too in a full new project with RN 0.58.0-rc.1

**Comment by mfferreira** [comment]
bump

**Comment by ryanhomer** [comment]
The various solutions lead to different results for me but I still haven't be able to make jest work properly. This is by far the most difficult RN upgrade to date due to the babel changes. Does anyone have any idea who is looking into this? Is it the RN, babel or jest team? It seems like we're all just locked in a room babbling to ourselves in isolation. 🙂

I would normally just wait but apparently RN 0.58.0 has a fix for the AAPT2 issue and we're trying to get this resolved since Google will soon not be allowing `android.enableAapt2=false` in `gradle.properties`.

>WARNING: The option 'android.enableAapt2' is deprecated and should not be used anymore.
Use 'android.enableAapt2=true' to remove this warning.
It will be removed at the end of 2018.

🌧 When it rains...

**Comment by yusufkaracin** [comment]
@rszalski's solution worked for me;

> 1. Create `babel.config.js` with the following content (basically equivalent to stock `.babelrc`)
> 
> ```
> module.exports = function (api) {
>   api.cache(true)
> 
>   return {
>     presets: ['module:metro-react-native-babel-preset']
>   }
> }
> ```
> 1. Remove `.babelrc`
> 2. Run `yarn add --dev babel-jest babel-core@^7.0.0-bridge.0 @babel/core`
> 
> I didn't have to add any transforms to jest after that.

**Comment by ryanhomer** [comment]
The closest I've gotten to a working Jest with that configuration gives the following errors at various points during the tests:

- ReferenceError: You are trying to `import` a file after the Jest environment has been torn down.
- TypeError: Cannot read property 'default' of undefined
- TypeError: stacktraceParser.parse is not a function

Additionally, it freezes during one of the tests and never completes. Also, there's nothing specific about the test it freezes on; if I remove that test, it'll get hung up on another.

I've been trying to get this resolved for weeks now. I don't know what else to try.

**Comment by clydedacruz** [comment]
adding the `transform` to my jest config in package.json fixed it for me
```
"jest": {
    "preset": "react-native",
    "transform": {
      "^.+\\.(js)$": "<rootDir>/node_modules/react-native/jest/preprocessor.js"
    }
  },
```

**Comment by nol13** [comment]
clydedacruz's fix breaks running tests from command line for me, that the above babel config fixed, but it does fix running tests from within webstorm :/

**Comment by nol13** [comment]
OK, current hopefully working solution: @rszalski's babel.config.js

\+

```"jest": {
    "preset": "react-native",
    "transform": {
      "^.+\\.js$": "babel-jest"
    },
```
the transform only required for running tests from within webstorm, from command line the babel config works fine by itself

**Comment by dsandor** [comment]
I just ran into this error message and found this thread.  Just putting this here in case anyone else has a similar issue.  I work on multiple projects under a `~/Projects` folder structure like `~/Projects/react-app-foo` and `~/Projects/some-node-app`.

A few days ago I accidentally copies some of my `.dotfiles` from a react-native app into my `~/Projects` folder.  Suddenly my node app was throwing the Couldn't find preset error when running jest on my node project.  Deleting the `.babelrc` from my parent folder fixed this for me.  Hope it helps someone else.

**Comment by gutoglup** [comment]
@dsandor but in this case I believe the problem is another. Because the problem reported in this topic is related to running the tests with Jest, and running the react-native project. I believe that the folder structure does not imply this error that is occurring, and the error logs that are being displayed.

**Comment by anubhavgupta** [comment]
Adding "postinstall": "rm $( find node_modules -name .babelrc)" in the scripts section of package.json and then running `npm install` OR `yarn` fixes this issue for me.

**Comment by bobowork** [comment]
> I seemed to have solved this by removing **everything** in the `babelrc` file:
> 
> ```
> # .babelrc
> {
> }
> ```
> `react-native-git-upgrade` then tries to re-add this during installation. I kept removing it and re-saving the file, and it continued without crashing. Then I added the contents back.

Confirmed @Amnesthesia 's solution works.

**Comment by liangguohun** [comment]
"jest": {
    "preset": "react-native"//do not contain this line maybe can help
  }

**Comment by methodbox** [comment]
I have also solved this by removing my .babelrc in the root (wasn't relevant anyway). 

But it seems Expo is not handling this correctly. Shouldn't we be able to have localized Babel configuration in the root of the app?

Maybe this requires a configuration in the ```.expo```?

**Comment by gutoglup** [comment]
> Adding `"postinstall": "rm $( find node_modules -name .babelrc)"` in the scripts section of package.json and then running `npm install` OR `yarn` fixes this issue for me.

I tried this, but not works.

**Comment by gutoglup** [comment]
> Adding `"postinstall": "rm $( find node_modules -name .babelrc)"` in the scripts section of package.json and then running `npm install` OR `yarn` fixes this issue for me.

> "jest": {
> "preset": "react-native"//do not contain this line maybe can help
> }

I tried all the solutions presented, but there was success in none at the moment.
The Jest or React-Native team could speak on this case.

**Comment by arled** [comment]
Why do people think this is down to upgrading?!! I've Just started a fresh project with the RN 0.57.8 and I have this issue straight out of the box. Has anyone got a real solution?

**Comment by arled** [comment]
Thank me later:

```
  "jest": {
    "preset": "react-native",
    "transform": { "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js" }
  }
```

**Comment by gutoglup** [comment]
> Thank me later:
> 
> ```
>   "jest": {
>     "preset": "react-native",
>     "transform": { "^.+\\.js$": "<rootDir>/node_modules/react-native/jest/preprocessor.js" }
>   }
> ```

I've tried that too.
If you create a clean project, can you run some test with the jest using the renderer?

Because of what has already been explained, either you put the preset to `react-native` and run the tests, or you run the application with the` module: react-native-babel-preset` preset. If you try to put both of them up, it's the problem that the community mentioned earlier in this issue.

**Comment by methodbox** [comment]
If you read this and related errors, you will notice that the direcotry it’s throwing the error from is inside node_modules.

A .babelrc will override in level of hierarchy other Babel configurations in subdirectories.

It would appear someone’s npm package is shipping with a .babelrc that presents a conflict when over-ridden.

This might also be another dot file in one of your project directories.

I would verify you don’t have another configuration file that might be read as a Babel config (I believe it accepts others than .babelrc)

As a test, try removing ALL dot files from your project (just move them if you need to save them), remove node_modules, then npm install again.

**Comment by Devil-Cong** [comment]
I met the same situation.
I initialized a project with `react-native-cli`, then I integrated `MboX` and use decorators, so my `package.json` and `.babelrc` like this:

```
"dependencies": {
    "mobx": "^4.8.0",
    "mobx-react": "^5.4.3",
    "react": "16.6.0-alpha.8af6728",
    "react-native": "0.57.4"
},
"devDependencies": {
    "@babel/plugin-proposal-decorators": "^7.2.2",
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.49.1",
    "react-test-renderer": "16.6.0-alpha.8af6728"
}
```
```
{
    "presets": ["module:metro-react-native-babel-preset"],
    "plugins": [
        ["@babel/plugin-proposal-decorators", { "legacy": true }]
    ]
}
```
When i run `react-native-git-upgrade`, i get this error:
> git-upgrade ERR! An error occurred during upgrade: 
git-upgrade ERR! Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.

Then i try to remove the `@babel/plugin-proposal-decorators` plugin from `.babelrc` like this:
```
{
  "presets": ["module:metro-react-native-babel-preset"]
}
```
I get another error:
> git-upgrade ERR! An error occurred during upgrade: 
git-upgrade ERR! Error: Couldn't find preset "module:metro-react-native-babel-preset" relative to directory...

Then i change `.babelrc` to `babel.config.js`:
```
module.exports = {
    presets: ['module:metro-react-native-babel-preset'],
    plugins: [
        ['@babel/plugin-proposal-decorators', { legacy: true }]
    ]
}
```
Run `react-native-git-upgrade`, it work!

I found that if I remove the `.babelrc` first and run `react-native-git-upgrade`, it also work!（need to put `.babelrc` back after upgraded.）

Hope this helps someone else.

**Comment by gutoglup** [comment]
@Devil-Cong I did the test according to what you missed, it did not work either. Still giving the same problem.

**Comment by Devil-Cong** [comment]
@gutoglup Sorry，but it work for me.
Did you try remove the `.babelrc` and run `react-native-git-upgrade`?

**Comment by Siva123Prasad** [comment]
When we try to run, this is what happens
bash: react-native-git-upgrade: command not found
@Devil-Cong

**Comment by Devil-Cong** [comment]
@Siva123Prasad You should install `react-native-git-upgrade` first. Look at [here](https://facebook.github.io/react-native/docs/upgrading).

**Comment by marhaupe** [comment]
I experienced the same issue for a while now. Yesterday, jest released jest@24.0.0, which upgraded its internals to Babel 7. I upgraded my dependency to the following:

`"jest": "24.0.0"`
`"babel-jest": "24.0.0"`
`"metro-react-native-babel-preset": "0.51.1"`
while still being on `"react-native": "0.57.8"`

I hope it works for some of you

**Comment by kelset** [comment]
It seems that this issue was caused by incorrect upgrading - please refer to @Devil-Cong's comment above for a good step by step of how to avoid/fix the issue: https://github.com/facebook/react-native/issues/21241#issuecomment-456088170

Closing & locking as that comment fixes it, and also we are working on [improving the upgrading procedure](https://github.com/react-native-community/discussions-and-proposals/issues/68)
